packages/swing-store/src/snapStore.js

```diff
@@ -293,33 +310,51 @@ export function makeSnapStore(
     );
   }
 
+  const sqlGetSnapshotInfo = db.prepare(`
+    SELECT endPos, hash, uncompressedSize, compressedSize
+    FROM snapshots
+    WHERE vatID = ?
+    ORDER BY endPos DESC
+    LIMIT 1
+  `);
+
+  function getSnapshotInfo(vatID) {
+    return /** @type {SnapshotInfo} */ (sqlGetSnapshotInfo.get(vatID));
+  }
+
+  const sqlDeleteVatSnapshots = db.prepare(`
+    DELETE FROM snapshots
+    WHERE vatID = ?
+  `);
+
   /**
-   * @param {string} hash
+   * @param {string} vatID
```

Comment:
let's add a comment: `delete all snapshots for a given vat, e.g. when the vat is terminated`